src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,175 @@
+        } else {
+            first = newNode;
+        }
+        currentNode.prev = newNode;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        if (list == null || list.size() == 0) {
+            throw new IllegalArgumentException("List is empty!");
+        }
+        for (T element : list) {
+            add(element);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> newNode = getNodeByIndex(index);
+        if (newNode == null) {
+            throw new IndexOutOfBoundsException("Index is bigger, than list size");
+        }
+        return newNode.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> newNode = getNodeByIndex(index);
+        if (newNode == null) {
+            throw new IndexOutOfBoundsException("Index was " + index + " and the size is "
+                    + size());
+        }
+        T oldValue = newNode.item;
+        newNode.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        var currentNode = getNodeByIndex(index);
+        if (currentNode == null) {
+            throw new IndexOutOfBoundsException("Index was " + index + " and the size is "
+                    + size());
+        }
+        if (index == 0) {
+            var nextCurrent = currentNode.next;
+            if (nextCurrent != null) {
+                nextCurrent.prev = null;
+            }
+            first = nextCurrent;
+            return currentNode.item;
+        }
+        if (currentNode.next == null) {
+            var prevCurrent = currentNode.prev;
+            prevCurrent.next = null;
+            return currentNode.item;
+        }
+        var prevCurrent = currentNode.prev;
+        prevCurrent.next = currentNode.next;
+        var nextCurrent = currentNode.next;
+        nextCurrent.prev = currentNode.prev;
+        return currentNode.item;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        var i = getIndexByValue(object);
+        if (i == -1) {
+            return false;
+        }
+        remove(i);
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        int i = 0;
+        for (Node<T> element = first; element != null; element = element.next) {
+            i++;
+        }
+        return i;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return first == null;
+    }
+
+    private Node<T> getNodeByIndex(int index) {
+        checkIndex(index);
+        int i = 0;
+        for (Node<T> element = first; element != null; element = element.next) {
+            if (i == index) {
+                return element;
+            }
+            i++;
+        }
+        return null;
```

Comment:
Implement the variable - size, if the index is index <= size / 2. You will start the search from the head. If index> size / 2 You start the search from the end. This will improve your search algorithm.